### PR TITLE
Minor: add test + docs for 2 argument trunc with columns

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Related to https://github.com/apache/arrow-datafusion/pull/6942 from @Syleechan 

# Rationale for this change
While reviewing https://github.com/apache/arrow-datafusion/pull/6942 I noticed the two argument version of `trunc` wasn't tested via sql and it wasn't documented

# What changes are included in this PR?

add test + docs for 2 argument trunc with columns

# Are these changes tested?
Yes

# Are there any user-facing changes?

### CODE REVIEW

**Comment by alamb** [comment]
@Syleechan  would you have time to review this PR?

File: datafusion/core/tests/sqllogictests/test_files/scalar.slt [hunk 1]
```diff
@@ -918,6 +918,15 @@ select trunc(4.267, 3), trunc(1.1234, 2), trunc(-1.1231, 6), trunc(1.2837284, 2)
...
 4.267 1.12 -1.1231 1.28 1
 
+# trunc with columns and precision
+query RRR rowsort
+select trunc(a, 3) as a3, trunc(a, 1) as a1, trunc(arrow_cast(a, 'Float64'), 3) as a3_f64 from small_floats;
+----
+-0.7 -0.7 -0.7
+-1 -1 -1
+0.2 0.2 0.2
+0.5 0.5 0.5
```
Comment (viirya) [inline datafusion/core/tests/sqllogictests/test_files/scalar.slt:928]: Hmm, as `a` has at most 1 decimal point, `trunc(a, 3)` or  `trunc(a, 1)` basically cannot show the difference. I think it doesn't actually test what we want?

File: docs/source/user-guide/sql/scalar_functions.md [hunk 2]
```diff
@@ -507,14 +507,17 @@ tanh(numeric_expression)
...
+trunc(numeric_expression[, decimal_places])
 ```
 
 #### Arguments
 
 - **numeric_expression**: Numeric expression to operate on.
   Can be a constant, column, or function, and any combination of arithmetic operators.
 
+- **decimal_places**: Optional. The number of decimal places to truncate to.
+  Defaults to 0.
```
Comment (viirya) [inline docs/source/user-guide/sql/scalar_functions.md:519]: Doc looks good to me. Just one question. Can `decimal_places` be negative?

File: datafusion/core/tests/sqllogictests/test_files/scalar.slt (hunk 1 above)
Comment (Syleechan) [reply datafusion/core/tests/sqllogictests/test_files/scalar.slt:928]: maybe add some long decimal point float number

File: docs/source/user-guide/sql/scalar_functions.md (hunk 2 above)
Comment (Syleechan) [reply docs/source/user-guide/sql/scalar_functions.md:519]: > 

can not be negative, modify the trunc function just meet the self-defined precision of decimal to '.'

**Review by Syleechan — COMMENTED** [review]
Thanks @alamb , I left some comments.

File: datafusion/core/tests/sqllogictests/test_files/scalar.slt (hunk 1 above)
Comment (alamb) [reply datafusion/core/tests/sqllogictests/test_files/scalar.slt]: Added

File: docs/source/user-guide/sql/scalar_functions.md (hunk 2 above)
Comment (alamb) [reply docs/source/user-guide/sql/scalar_functions.md]: TIL turns out that negative precision means "truncate to the left of the decimal place": https://www.postgresqltutorial.com/postgresql-math-functions/postgresql-trunc/#:~:text=If%20the%20precision%20argument%20is,The%20precision%20argument%20is%20optional. 

```
postgres=# select trunc(12345.6789, -2);
 trunc
-------
 12300
(1 row)
```

This is actually what the current datafusion implementation does too so I have added a test and updated the docs

**Comment by alamb** [comment]
Thanks @viirya  and @Syleechan  - I have addressed your comments.

**Review by Syleechan — APPROVED** [review]
Thanks, Approved.

**Review by viirya — APPROVED** [review]

**Pull request merged**